watch.go

```diff
@@ -103,48 +121,54 @@ func (fb *Firebase) Watch(notifications chan Event) error {
 			// 		event: put
 			// 		data: {"path":"/","data":{"foo":"bar"}}
 
-			var evt []byte
-			var dat []byte
-			isPrefix := true
-			var result []byte
-
-			// For possible results larger than 64 * 1024 bytes (MaxTokenSize)
-			// we need bufio#ReadLine()
-			// 1. step: scan for the 'event:' part. ReadLine() oes not return the \n
-			// so we have to add it to our result buffer.
-			evt, isPrefix, scanErr = scanner.ReadLine()
+			payload := make([]string, 2)
+			scanErr = func(payload []string) error {
```

Comment:
It is indeed an option. The closure is for me a way to avoid the "break"
with a label. which I don't really like; but this is probably a question of
personal style.

The other benefits of my solution is that it handles correctly the case
where a reply has more than 2 lines. Perhaps this is what causes the
confusion.

█  Jacques

On 31 July 2016 at 20:54, Steven Berlanga <notifications@github.com> wrote:

> In watch.go
> <https://github.com/zabawaba99/firego/pull/50#discussion_r72908852>:
>
> > @@ -103,48 +121,54 @@ func (fb *Firebase) Watch(notifications chan Event) error {
> >  			// 		event: put
> >  			// 		data: {"path":"/","data":{"foo":"bar"}}
> >
> > -			var evt []byte
> > -			var dat []byte
> > -			isPrefix := true
> > -			var result []byte
> > -
> > -			// For possible results larger than 64 * 1024 bytes (MaxTokenSize)
> > -			// we need bufio#ReadLine()
> > -			// 1. step: scan for the 'event:' part. ReadLine() oes not return the \n
> > -			// so we have to add it to our result buffer.
> > -			evt, isPrefix, scanErr = scanner.ReadLine()
> > +			payload := make([]string, 2)
> > +			scanErr = func(payload []string) error {
>
> The logic inside of the closure is a little confusing. I like the idea of
> enforcing that we should only ever get two lines from the connection. What
> about something like
>
> var (
>     lineCount int
>     line      string
> )
> line, scanErr = scanner.ReadString('\n')for ; lineCount != len(payload); line, scanErr = scanner.ReadString('\n') {
>     if scanErr != nil {
>         break scanning
>     }
>
>     line = strings.Trim(line, " \r\n")
>     if len(line) == 0 { // empty line
>         scanErr = errors.New("Bad formated body")
>         break scanning
>     }
>     payload[lineCount] = line
>     lineCount++
> }
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/zabawaba99/firego/pull/50/files/91637d68c23bcd6761930b23882b18621b759365#r72908852>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAB2iL5RvdcFrr5J1iIEVGZLhtqFEygQks5qbO94gaJpZM4JYHTS>
> .
>